src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -21,27 +21,27 @@ private enum SingleBufferHandleState : byte { None, InProcess, Set }
         // BufferList property variables.
         // Note that these arrays are allocated and then grown as necessary, but never shrunk.
         // Thus the actual in-use length is defined by _bufferListInternal.Count, not the length of these arrays.
-        private WSABuffer[] _wsaBufferArray;
-        private GCHandle[] _multipleBufferGCHandles;
+        private WSABuffer[]? _wsaBufferArray;
+        private GCHandle[]? _multipleBufferGCHandles;
 
         // Internal buffers for WSARecvMsg
-        private byte[] _wsaMessageBuffer;
+        private byte[]? _wsaMessageBuffer;
         private GCHandle _wsaMessageBufferGCHandle;
-        private byte[] _controlBuffer;
+        private byte[]? _controlBuffer;
         private GCHandle _controlBufferGCHandle;
-        private WSABuffer[] _wsaRecvMsgWSABufferArray;
+        private WSABuffer[]? _wsaRecvMsgWSABufferArray;
         private GCHandle _wsaRecvMsgWSABufferArrayGCHandle;
 
         // Internal SocketAddress buffer
         private GCHandle _socketAddressGCHandle;
-        private Internals.SocketAddress _pinnedSocketAddress;
+        private Internals.SocketAddress? _pinnedSocketAddress;
 
         // SendPacketsElements property variables.
-        private FileStream[] _sendPacketsFileStreams;
+        private FileStream[]? _sendPacketsFileStreams;
 
         // Overlapped object related variables.
-        private PreAllocatedOverlapped _preAllocatedOverlapped;
-        private readonly StrongBox<SocketAsyncEventArgs> _strongThisRef = new StrongBox<SocketAsyncEventArgs>(); // state for _preAllocatedOverlapped; .Value set to this while operations in flight
+        private PreAllocatedOverlapped _preAllocatedOverlapped = null!; // initialized by helper called from ctor
```

Comment:
I know this is the standard way of forcing non-nullability of fields, but I think it might become an issue if/when new constructors are being added or modified: the compiler will provide no warnings if the field is not being initialized. Do you think it might make sense to move those forced initializations into the individual constructors that need it? It would also make it easier to audit in the future. cc @buyaa-n @stephentoub

Reply:
The right way to handle this will be with the `[MemberNotNull]` attribute that's coming.  I'd prefer to wait for that rather than changing the scheme we employ now.  I also don't think the duplication of all that initialization would be worthwhile.